src/device-state/current-state.ts

```diff
@@ -64,7 +56,20 @@ const stripDeviceStateInLocalMode = (state: DeviceStatus): DeviceStatus => {
 	};
 };
 
-const report = async ({ stateDiff, conf }: StateReport) => {
+/**
+ * Try to patch difference in device state
+ *
+ * report attempts to patch the difference in device state from the last
+ * report successfully sent.
+ */
+async function report(stateDiff: Partial<DeviceStatus>): Promise<void> {
+	const conf = await config.getMany([
```

Comment:
I sort of liked that this function was unaware of the `config` module and it just got the configs as parameter. I think it makes it cleaner but that's just an opinion really

Reply:
I agree, it also opened up the possibility of only querying for these values on startup and then reacting to them changing rather than querying them on every single report attempt (a bunch/all of them hit the disk right? and could fail/be slow due to that if things are busy)